pkg/plugins/runtime/k8s/controllers/inbound_converter.go

```diff
@@ -131,7 +131,29 @@ func inboundForServiceless(zone string, pod *kube_core.Pod, name string, nodeLab
 	}
 }
 
+// Deprecated: LegacyInboundInterfacesFor is currently only used for delegated gateway and Mesh without MeshService exclusive
+// to not change order of inbounds.
+// For gateway we pick first inbound to take tags from. Delegated gateway identity relies on this.
+// For Dataplanes when MeshService is disabled we base identity and routing
+// TODO: We should revisit this when we rework identity. More in https://github.com/kumahq/kuma/issues/3339
+func (i *InboundConverter) LegacyInboundInterfacesFor(ctx context.Context, zone string, pod *kube_core.Pod, services []*kube_core.Service) ([]*mesh_proto.Dataplane_Networking_Inbound, error) {
+	return i.inboundInterfacesFor(ctx, zone, pod, services)
+}
+
+// InboundInterfacesFor should be used when MeshService mode is Exclusive
 func (i *InboundConverter) InboundInterfacesFor(ctx context.Context, zone string, pod *kube_core.Pod, services []*kube_core.Service) ([]*mesh_proto.Dataplane_Networking_Inbound, error) {
+	inbounds, err := i.inboundInterfacesFor(ctx, zone, pod, services)
+	if err != nil {
+		return inbounds, err
+	}
+
+	// Right now we will build multiple inbounds for each service selecting port, but later on
```

Comment:
I'm a bit lost with "right now" and "later on", by "later" you mean like "later when Envoy receives the config"? Or "later" when we develop some other feature?

Reply:
I agree. The comment will be persist here for long, so we should serve a future reader's perspective.

Reply:
Removed the whole comment as it was outdated. I've updated comment for the whole method